src/v/cluster/data_migration_backend.cc

```diff
@@ -892,12 +928,26 @@ ss::future<> backend::abort_all_topic_work() {
         vlog(dm_log.info, "one topic work state completed");
     }
 }
-
-ss::future<errc> backend::create_topic(
-  const model::topic_namespace& local_nt,
+ss::future<
+  result<std::reference_wrapper<const cloud_storage::topic_manifest>, errc>>
+backend::maybe_download_topic_manifest(
+  const model::topic_namespace& nt,
   const std::optional<model::topic_namespace>& original_nt,
   const std::optional<cloud_storage_location>& storage_location,
-  retry_chain_node& rcn) {
+  tsws_lwptr_t tsws) {
+    if (tsws->cached_topic_manifest()) {
+        vlog(
+          dm_log.trace,
+          "using cached topic manifest for topic {} (location {})",
+          original_nt.value_or(nt),
+          storage_location);
+        co_return *tsws->cached_topic_manifest();
```

Comment:
can't we return a dangling reference here, due to scheduling points?

Reply:
Change this to return a value